jekyll/_cci2/rerun-failed-tests-only.adoc

```diff
@@ -0,0 +1,275 @@
+
+[#configure-a-job-running-ruby-cucumber-tests]
+=== Configure a job running Ruby (Cucumber) tests
+
+. Modify your test command to look something similar to:
++
+```yaml
+- run: mkdir -p ~/cucumber
+- run:
+    command: |
+    circleci tests glob features/**/*.feature | circleci tests run --command="xargs bundle exec cucumber --format junit --out ~/cucumber/junit.xml" --verbose
+```
+
+. Update the `glob` command to match your use case. See the Cucumber section in the xref:collect-test-data#cucumber[Collect Test Data] document for details on how to output test results in an acceptable format for `Cucumber`.
+
+[#configure-a-job-running-cypress-tests]
+=== Configure a job running Cypress tests
+
+. Install the `cypress-multi-reporters` and `mocha-junit-reporter` dependencies. If using `npm`, run the following on your local machine:
++
+```bash
+npm install --save-dev cypress-multi-reporters mocha-junit-reporter
+```
++
+Your `package.json` / `package-lock.json` will be updated and committed on your next push from that branch. 
+
+. Create and set up the reporter config file, if it does not already exist. In this example, the file is named `reporter-config.json`.
++
+```json
+{
+  "reporterEnabled": "spec, mocha-junit-reporter", // set the reporters
+  "reporterOptions": {
+    "mochaFile": "results/junit/junit-[hash].xml", // each suite produces its own junit :(, save them with unique hash
+  }
+}
+```
+
+. Modify your test command to use the two reporter flags and `circleci tests run`:
++
+```yaml
+     -run:
+        name: run tests
+        command: | 
+          cd ./cypress 
+          npm ci 
+          npm run start &
+          circleci tests glob "cypress/**/*.cy.js" | circleci tests run --command="xargs npx cypress run --reporter cypress-multi-reporters --reporter-options configFile=reporter-config.json --spec" --verbose
+```
++
+Remember to modify the `glob` command for your specific use case.
+
+. Because Cypress does not output the expected `filename` attribute on its JUnit XML files, follow the steps outlined link:https://github.com/michaelleeallen/mocha-junit-reporter/issues/132[in this issue] to massage the test results into the proper format. In this example, we have saved a copy of the script to a file named `fix-junit.js`. 
++
+You can then invoke this script by adding a new `run` step in your `.circleci/config.yml` file. The snippet below also includes the `store_test_results` step to upload test results:
++
+```yaml
+    - run:
+       when: always
+       name: process test results (add in file path in junit)
+       command: |
+          cd ./cypress
+          node ./scripts/fix-junit.js
+     - store_test_results: 
+       path: ./cypress/results
+```  
++
+Your new testing job's `.circleci/config.yml` definition will have both snippets from steps 3 and 4 above, one right after the other.
```

Comment:
Do we want to say "should have both snippets" as opposed to "will have"?

Reply:
yeah also a good idea